WordPress/gutenberg
# Parsing: Replace server serialization by parsed outerHTML

Related: #2806
Relies on: https://github.com/nylen/phpegjs/pull/3 (changes manually effected here, but will be lost in future parser revisions)

This pull request seeks to change the behavior of selective `wpautop` application, simplifying the behavior to preserving the original markup of block content returned from the parser, rather than reserializing the parsed block node. In particular, this will be necessary for block nesting since, as implemented, the server is not capable of reserializing blocks with nested blocks if the block's save behavior is implemented on the client.

__Testing instructions:__

Ensure that tests pass:

```
npm test
```
```
phpunit
```

Repeat testing instructions from #2806
